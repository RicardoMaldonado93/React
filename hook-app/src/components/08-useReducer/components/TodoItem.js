import React from 'react'

export const TodoItem = ({ todo, index, handleDelete, handleToggle }) => {

    // console.log("todo item")
    return (

        <li
            className="list-group-item"
            key={todo.id}
        >

            <p
                className={`${todo.done && 'complete'}`}
                onClick={() => handleToggle(todo.id)}
            >
                {index + 1}. {todo.desc}
            </p>

            <button
                className="btn btn-danger"
                onClick={() => handleDelete(todo.id)}
            >
                Borrar
            </button>
        </li>
    )
}
