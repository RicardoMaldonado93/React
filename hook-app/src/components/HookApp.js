import React, { Component } from 'react';

const HookApp = () =>{
 
    return (
        <div className={"container-fluid"}>
         <h1>Hola Mundo</h1>   
        </div>
    )
}

export default HookApp;
