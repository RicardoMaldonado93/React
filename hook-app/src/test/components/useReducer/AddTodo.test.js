import React from 'react';
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { demoTodos } from '../../fixtures/demoTodos';
import { AddTodo } from '../../../components/08-useReducer/components/AddTodo';

describe('Unitary test for <AddTodo />', () => {
    const handleAddTodo = jest.fn();
    const wrapper = shallow( 
        <AddTodo 
            handleAddTodo = { handleAddTodo }
        /> 
    );

    test('should be shown correctly ', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('dont should call the function handleAddTodo', () => {
        const formSubmit = wrapper.find("form").prop("onSubmit");

        formSubmit({ preventDefault(){ } });

        expect( handleAddTodo ).toBeCalledTimes(0); 
    });

    test('should call the function handleAddTodo', () => {
        const value = "Aprender SQL";
        wrapper.find("input").simulate("change",{
            target:{
                value,
                name: "description"
            }
         })

        const formSubmit = wrapper.find("form").prop("onSubmit");
        formSubmit({ preventDefault(){ } });

        expect( handleAddTodo ).toBeCalledTimes(1); 
        expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object) ); 
        expect( handleAddTodo ).toHaveBeenCalledWith( { 
            desc: value,
            done: false,
            id: expect.any( Number )
         } ); 
         
        expect( wrapper.find("input").prop("value") ).toBe("");
    });
    
})

