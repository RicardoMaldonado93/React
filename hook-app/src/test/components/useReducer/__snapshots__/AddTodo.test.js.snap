// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitary test for <AddTodo /> should be shown correctly  1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      style={
        Object {
          "margin": "0",
        }
      }
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
